src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -38,6 +38,8 @@ public class JedisClusterInfoCache {
   private final Map<String, ConnectionPool> nodes = new HashMap<>();
   private final ConnectionPool[] slots = new ConnectionPool[Protocol.CLUSTER_HASHSLOTS];
   private final HostAndPort[] slotNodes = new HostAndPort[Protocol.CLUSTER_HASHSLOTS];
+  private final Map<Integer, List<ConnectionPool>> replicaSlots = new HashMap<>(Protocol.CLUSTER_HASHSLOTS, 1.0f);
+  private final Map<Integer, List<HostAndPort>> replicaSlotNodes = new HashMap<>(Protocol.CLUSTER_HASHSLOTS, 1.0f);
```

Comment:
Use array instead of Map AND create + initialize array only when read-only is set.
Can keep the inner List.

Reply:
done